src/Ads/AssetSuggestionsService.php

```diff
@@ -0,0 +1,211 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Ads;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\WP;
+
+/**
+ * Class AssetSuggestionsService
+ *
+ * Suggest assets and possible final URLs.
+ *
+ * @since x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Ads
+ */
+class AssetSuggestionsService implements Service {
+
+	/**
+	 * AssetSuggestionsService constructor.
+	 *
+	 * @param WP $wp
+	 */
+	public function __construct( WP $wp ) {
+		$this->wp = $wp;
+	}
+
+	/**
+	 * Get posts that can be used to suggest assets
+	 *
+	 * @param string $search The search query.
+	 * @param int    $per_page Number of items per page.
+	 * @param int    $offset Used in the get_posts query.
+	 *
+	 * @return array
+	 */
+	protected function get_post_suggestions( $search, $per_page, $offset = 0 ): array {
+		$post_suggestions    = [];
+		$excluded_post_types = [ 'attachment' ];
+
+		$post_types = $this->wp->get_post_types(
+			[
+				'exclude_from_search' => false,
+				'public'              => true,
+			]
+		);
+
+		// Exclude attachment post_type
+		$filtered_post_types = array_diff( $post_types, $excluded_post_types );
+
+		$args = [
+			'post_type'      => $filtered_post_types,
+			'posts_per_page' => $per_page,
+			'post_status'    => 'publish',
+			's'              => $search,
+			'offset'         => $offset,
+		];
+
+		$posts = $this->wp->get_posts( $args );
+
+		foreach ( $posts as $post ) {
+			$post_suggestions[] = $this->format_final_url_response( $post->ID, 'post', $post->post_title, get_permalink( $post->ID ) );
+		}
+
+		return $post_suggestions;
+	}
+
+	/**
+	 * Get terms that can be used to suggest assets
+	 *
+	 * @param string $search The search query
+	 * @param int    $per_page Number of items per page
+	 *
+	 * @return array
+	 */
+	protected function get_terms_suggestion( $search, $per_page ): array {
```

Comment:
💅 

Minor, but just noticed that `get_terms_suggestion` is in singular. Shouldn't be `suggestions`?